Webapp/sources/blueprints/reaction_list/routes.py

```diff
@@ -85,25 +86,92 @@ def get_reactions() -> Response:
     return jsonify({"reactionDetails": reaction_details})
 
 
-@reaction_list_bp.route("/get_schemata", methods=["GET", "POST"])
+@reaction_list_bp.route("/get_reaction_images", methods=["GET", "POST"])
+@login_required
+def get_reaction_images() -> Response:
+    """
+    Gets a list of reaction images for the active workbook.
+    """
+    workbook_name = str(request.form.get("workbook"))
+    workgroup_name = str(request.form.get("workgroup"))
+    workbook = services.workbook.get_workbook_from_group_book_name_combination(
+        workgroup_name, workbook_name
+    )
+    abort_if_user_not_in_workbook(workgroup_name, workbook_name, workbook)
+    sort_crit = str(request.form.get("sortCriteria"))
+    reaction_list = services.reaction.list_active_in_workbook(
+        workbook_name, workgroup_name, sort_crit
+    )
+    reaction_images = services.reaction.make_reaction_image_list(reaction_list)
+    return {"reaction_images": reaction_images, "sort_crit": escape(sort_crit)}
+
+
+@reaction_list_bp.route("/get_smiles", methods=["GET", "POST"])
+@login_required
+def get_smiles() -> Response:
+    """
+    Gets a list of reaction smiles for the active workbook.
+    """
+    # must be logged in
+    workbook_name = str(request.form.get("workbook"))
+    workgroup_name = str(request.form.get("workgroup"))
+    workbook = services.workbook.get_workbook_from_group_book_name_combination(
+        workgroup_name, workbook_name
+    )
+    abort_if_user_not_in_workbook(workgroup_name, workbook_name, workbook)
+    sort_crit = str(request.form.get("sortCriteria"))
+    reaction_list = services.reaction.list_active_in_workbook(
+        workbook_name, workgroup_name, sort_crit
+    )
+    smiles_list = [r.reaction_smiles for r in reaction_list]
+    return {"smiles": smiles_list, "sort_crit": escape(sort_crit)}
+
+
+@reaction_list_bp.route("/get_rxns", methods=["GET", "POST"])
 @login_required
-def get_schemata() -> Response:
+def get_rxns() -> Response:
     """
-    Gets a list of reaction schemes for the active workbook. Images made from reaction SMILES using rdMolDraw2D (RDKit)
+    Gets a list of reaction RXNs for the active workbook.
     """
-    workbook_name = str(request.form["workbook"])
-    workgroup_name = str(request.form["workgroup"])
+    # must be logged in
+    workbook_name = str(request.form.get("workbook"))
+    workgroup_name = str(request.form.get("workgroup"))
+    workbook = services.workbook.get_workbook_from_group_book_name_combination(
+        workgroup_name, workbook_name
+    )
+    abort_if_user_not_in_workbook(workgroup_name, workbook_name, workbook)
```

Comment:
Couple of lines in this file that could be replaced by a decorator again https://github.com/AI4Green/AI4Green/blob/main/Webapp/sources/decorators.py

Reply:
the decorator says it may fail for functions returning json to frontend - would that be a problem here?

Reply:
I think it should be ok for get_rxns, might cause an issue for save_new_images though.